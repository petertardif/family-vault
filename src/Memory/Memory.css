.Memory {
    margin: 5px auto;
    opacity: 0.8;
    border-radius: 13px;
    padding: 1em;
    display: flex;
    justify-content: center;
    border: 1px gray solid;
}

@media only screen and (min-width: 320px) {
    .Memory {
        width: 75%;
        min-width: 300px;
        padding: 10px;
    }
}

@media only screen and (min-width: 540px) {
    .Memory {
        width: 65%;
    }
}

@media only screen and (min-width: 800px) {
    .Memory {
        width: 55%;
    }
}