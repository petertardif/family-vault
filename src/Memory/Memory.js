import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import Toggle from '../Toggle/Toggle';
import { API_BASE_URL } from '../config';
import './Memory.css';
import { MemoryContext } from '../MemoryContext';

class Memory extends Component {
  // static defaultProps = {
  //   onDeleteMemory: () => {},
  // }
  static contextType = MemoryContext;

  handleDeleteRequest = (event) => {
    event.preventDefault()
    const memoryId = this.props.id
    debugger
    fetch(`${API_BASE_URL}/memories/${memoryId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(() => {
        this.context.deleteMemory(memoryId)
        this.props.history.push(`/memorylist/${memoryId}`)
      })
      .catch(error => {
        console.error({ error })
      })
      this.props.deleteMemory(memoryId)
      debugger
      this.props.history.push(`/memorylist`)
      // this.props.history.push(`/memorylist/${memoryId}`)
    };
  

  render() {
    return (
      <li key={this.props.id} className='Memory'>
        <div className='Memory_row'>
          <h2 className='Memory_title'>
            <Link to={`/memory/${this.props.id}`}>
              {this.props.memory_title}
              </Link>
          </h2>
          <div className='Memory_dates'>
            <div className='Memory_dates-modified'>
                Modified <span className='Date'>{this.props.memory_date}</span> 
            </div>
          </div>
          <Toggle>
            {({ on, toggle }) => (
              <>
                {on && 
                  <>
                    <img src={this.props.media_url} alt={`supporting visual for ${this.props.title}`}/>
                    <p className='Memory_description'>
                      {this.props.memory_desc}
                    </p>
                    <div className='Memory_buttons'>
                      {/* <button
                        className='Btn_memory_edit'
                        // onClick={() => props.onClickEdit(props.id)}
                      >
                        Edit
                      </button> */}
                      <button
                        type='button'
                        className='Btn_memory_delete'
                        onClick={this.handleDeleteRequest}
                      >
                        Delete
                      </button>
                    </div>
                  </>
                }
                <button onClick={toggle}>Expand / Collapse</button>
              </>
            )}
          </Toggle>
        </div>
      </li>
    )
  }
}

Memory.defaultProps = {
  onClickDelete: () => {},
}

export default withRouter(Memory);