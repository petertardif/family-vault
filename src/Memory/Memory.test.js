import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Memory from './Memory'

describe(`Memory component`, () => {
  const props = {
    id: 'a',
    memory_title: 'test-class-name',
    memory_date: new Date(2018, 12, 15),
  }

  it('renders a .Memory by default', () => {
    const wrapper = shallow(<Memory />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Memory given props', () => {
    const wrapper = shallow(<Memory {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
