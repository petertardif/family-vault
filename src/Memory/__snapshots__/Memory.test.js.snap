// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memory component renders a .Memory by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Memory component renders the Memory given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
